lib/ecto/association.ex

```diff
@@ -233,11 +234,15 @@ defmodule Ecto.Association.Has do
         "association #{inspect name}, please set the :references option accordingly"
     end
 
-    assoc = Keyword.fetch!(opts, :queryable)
+    queryable = Keyword.fetch!(opts, :queryable)
 
-    unless is_atom(assoc) do
-      raise ArgumentError, "association queryable must be a model, got: #{inspect assoc}"
-    end
+    assoc =
+      case queryable do
```

Comment:
Yeah, I have this idea, but not sure if we need it in `BelongTo`. I'll refactor this :)